### ISSUE TITLE
Possible Text Sanitization Issue

### ISSUE THREAD

**Issue by chris-heney** [AntonOsika/gpt-engineer#871]
### The Error

Uneducated best guess: Unsanitized string in "clarification response" in ai.py:216 when trying to deserialize messages... maybe?  

**Background:**  I've never attempted to contribute before, and I don't know Python.  I'm just trying to help (even though I don't enjoy giving up the prompt I was using as it is for a personal project on [watergis.org](https://watergis.org) ... but since it's an open project anyway, I figured I would share every detail, including my exact prompt & responses.  I am trying to follow the submission instructions the best that I can.

```
  File "/Users/ringzer0/anaconda3/envs/GIS/bin/gpt-engineer", line 8, in <module>
    sys.exit(app())
             ^^^^^

  File "/Users/ringzer0/projects/gpt-engineer/gpt_engineer/main.py", line 96, in main
    messages = step(ai, dbs)
               ^^^^^^^^^^^^^

  File "/Users/ringzer0/projects/gpt-engineer/gpt_engineer/steps.py", line 192, in gen_clarified_code
    messages = AI.deserialize_messages(dbs.logs[clarify.__name__])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/ringzer0/projects/gpt-engineer/gpt_engineer/ai.py", line 216, in deserialize_messages
    return list(messages_from_dict(json.loads(jsondictstr)))  # type: ignore
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/ringzer0/anaconda3/envs/GIS/lib/python3.11/site-packages/langchain/schema/messages.py", line 417, in messages_from_dict
    return [_message_from_dict(m) for m in messages]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/ringzer0/anaconda3/envs/GIS/lib/python3.11/site-packages/langchain/schema/messages.py", line 417, in <listcomp>
    return [_message_from_dict(m) for m in messages]
            ^^^^^^^^^^^^^^^^^^^^^

  File "/Users/ringzer0/anaconda3/envs/GIS/lib/python3.11/site-packages/langchain/schema/messages.py", line 405, in _message_from_dict
    raise ValueError(f"Got unexpected message type: {_type}")

ValueError: Got unexpected message type: AIMessageChunk
```

### System

- MacOS
- M2 Max (32 Core)
- 96GB Ram

### Steps to Reproduce

- `git clone https://github.com/AntonOsika/gpt-engineer.git`
- `cd gpt-engineer`
- `conda create --name myenv python=3.11.5`
- `python -m pip install -e .`
- create .env file per documentation
- `mkdir spatial-nexus`
- `cd spatial-nexus`
- created file: prompt (see below)
- `cd ..`
- `gpt-engineer spatial-nexus`

 
### The Prompt

```
Create a docker-compose.yml composed of a job scheduling/queuing/running service, a PostGIS Database, capable of email and sms notifications.
There should also be a UI that can run in the browser comprised of HTML, CSS and JavaScript to handle submission requests.
A user should be able to use the UI to submit a "file conversion" request, submission requests will allow the user to upload a shape file.
There should be a URL that will reference the file conversion request/job/task/file/PostGIS-Record.
A user should be able to choose to list all of their uploaded files, and choose to download the file and specify the file format and coordinate reference system (CRS).
The system should use PostGIS and/or a combination of geopanda, rasterio, and/or gdal to facilitate the request.
Response should be represented by a webserver that includes a MIME type 'application/force-download'
```

### Responses

```
Summary of areas that need clarification:

1. The specific job scheduling/queuing/running service to be used in the docker-compose.yml.
2. The specific requirements for the PostGIS Database.
3. The mechanism for email and SMS notifications.
4. The specific requirements for the UI, including how it should handle submission requests.
5. The process for a user to submit a "file conversion" request and upload a shape file.
6. The structure of the URL that will reference the file conversion request/job/task/file/PostGIS-Record.
7. The process for a user to list all of their uploaded files, and choose to download the file and specify the file format and coordinate reference system (CRS).
8. The specific use of PostGIS and/or a combination of geopanda, rasterio, and/or gdal to facilitate the request.
9. The specific requirements for the webserver response, including the inclusion of a MIME type 'application/force-download'.

Clarifying question:

Could you please provide more details on the specific job scheduling/queuing/running service you want to be used in the docker-compose.yml?
Any that is compatible and will work well with the docker-compose configuration and/or the libraries and tools chosen.

Could you please elaborate on the specific requirements for the PostGIS Database?
The PostGIS database must work with the docker-compose.yml file, and represent the data that would need to be added and extracted from the database.
  
Could you please provide more details on the mechanism for email and SMS notifications?
Any, each service should be optional, easy to toggle on or off, and be represented independently in the docker-compose.yml file.
    
Could you please elaborate on the specific requirements for the UI, including how it should handle submission requests?
UI should prompt the user to Add a file to their file library, be able to list files in their library, and choose a download icon that will bring them to a download page for that resource/asset.

Could you please explain the process for a user to submit a "file conversion" request and upload a shape file?
A user would click "browse" that would open up their system's file selection dialog, where they could select their GIS/Shape-files.

Could you please describe the structure of the URL that will reference the file conversion request/job/task/file/PostGIS-Record?
Example (let's decide on calling it a "resource"): The URL structures should look like: /resource/add /resource/{resource_id} /resource/{resource_id}/download?format={file_format}&crs={crs_format} and /resource/list

Could you please explain the process for a user to list all of their uploaded files, and choose to download the file and specify the file format and coordinate reference system (CRS)?
While on the list page, the user should be able to click a "download" icon/button that will ask the user to supply (via dropdown) value for {file_format} (what kind of file format is required), and {crs_format}.

Could you please provide more details on the specific use of PostGIS and/or a combination of geopanda, rasterio, and/or gdal to facilitate the request?
PostGIS should be the spatial database used to store the user's uploaded resources/geospatial data.  The combination of the other libraries may be used to facilitate file conversion requests based on the {file_format} and {crs_format} requested.

Could you please clarify the specific requirements for the webserver response, including the inclusion of a MIME type 'application/force-download'?
In short, when the user submits the resource, file_format and crs_format, in a download request, the web server should respond in a way that let the browserdownload the file.

Nothing more to clarify.
```

Then the error appeared.
```

**Comment by ATheorell** [comment]
Hi, thanks for the very neat bug report. This seems to be the same problem as in #859 .
I believed this was fixed by degrading langchain to below 0.0.312, but maybe its not? Which version of langchain are you using? 
A little context: A problem here is that the problem occurs deeper inside langchain, so we don't see an easy fix on our side.

Interesting project!
